doc/src/guides/solving/solve-diophantine-equation.md

```diff
@@ -0,0 +1,207 @@
+# Solve a Diophantine Equation Algebraically
+
+Use SymPy to solve a Diophantine equation (find integer solutions to a
+polynomial equation) algebraically, returning a parameterized general solution
+if possible. For example, solving the [Pythagorean
+theorem](https://en.wikipedia.org/wiki/Pythagorean_theorem) $a^2 + b^2 = c^2$
+yields $(a=2pq, b=p^2-q^2, c=p^2-q^2)$.
```

Comment:
How about ", where $p,q \in \mathbb{Z}$ parameterize the infinite set of [Pythagorean triples](https://en.wikipedia.org/wiki/Pythagorean_triple)"

Reply:
Specifically, make it clear that `p` and `q` are new parameters introduced in the solution, which can take on any integer value to parameterize the full set of solutions.